src/parser.y

```diff
@@ -536,21 +544,23 @@ ImportWhat:
   jv_free(v);
 } |
 "include" ImportFrom {
-  jv v = block_const($2);
-  $$ = gen_import(jv_string_value(v), NULL, 0);
-  block_free($2);
-  jv_free(v);
+  if (block_is_error($2)) {
```

Comment:
@dtolnay, @wtlangford Note that this is a hack to prevent an assertion a few lines below in `block_const()`.  This is what I mean by cascading of error handling.  This becomes necessary because the `ImportFrom` rule below preserves the offending non-constant `block` so that `--debug-dump-block` can produce something useful.  Of course, without `gen_location()` that won't be so useful, but one of the next steps is to make all error cases also call gen_location().